docs/types/intersection-types.md

```diff
@@ -42,17 +42,61 @@ Complex.from (that:Float) = Complex.Num that 0
 ```
 and uses them to create all the values the _intersection type_ represents.
 
+> [!NOTE]
+> Note that if a `Float.from (that:Complex)` conversion were available in the scope, 
+> any `Complex` instance would be convertible to `Float` regardless of how it was constructed. 
+> To ensure that such mix-ins are only available on values that opt-in to being 
+> an intersection type (like in the `Complex` example above where we include 
+> the `Float` mix-in only if `self.im == 0`), we need to ensure that the conversion 
+> used to create the intersection type is not available in the default 
+> conversion resolution scope. Thus it cannot be defined in the same module 
+> as `Complex` or `Float` types, but instead it should be defined in a separate 
+> module that is only imported in the place that will be constructing the multi-values.
```

Comment:
That's correct and exactly as [prototyped in the tests](https://github.com/enso-org/enso/commit/3d8a0e1b90b20cfdfe5da8d2d3950f644a4b45b8#diff-c6ef852899778b52ce6a11ebf9564d102c273021b212a4848b7678e120776287R23).